pkg/liquidity-mining/contracts/fee-distribution/FeeDistributor.sol

```diff
@@ -278,9 +276,23 @@ contract FeeDistributor is IFeeDistributor, ReentrancyGuard {
             _tokenStartTime[token] = _roundDownTimestamp(block.timestamp);
         } else {
             timeSinceLastCheckpoint = block.timestamp - lastTokenTime;
-            if (!force && timeSinceLastCheckpoint < _TOKEN_CHECKPOINT_DEADLINE) {
-                // We can prevent a lot of SSTORES by only checkpointing tokens at a minimum interval
-                return;
+
+            if (!force) {
+                // Checkpointing N times within a single week is completely equivalent to checkpointing once at the end
+                // We then want to get as close as possible to a single checkpoint per week at Thurs 00:00 UTC.
+
+                // We then skip checkpointing if we're in the same week as the previous checkpoint.
+                bool alreadyCheckpointedThisWeek = _roundDownTimestamp(block.timestamp) ==
+                    _roundDownTimestamp(lastTokenTime);
+                // However we want to ensure that all of this week's fees are assigned to the current week without
+                // overspilling into the next week. To mitigate this, we checkpoint if we're near the end of the week.
+                bool nearingEndOfWeek = _roundUpTimestamp(block.timestamp) - block.timestamp < 1 days;
+
+                // This ensures that we get a single checkpoint for the beginning of the week and then multiple
+                // checkpoints to give an accurate reading of the token balance at the end of the week.
+                if (alreadyCheckpointedThisWeek && !nearingEndOfWeek) {
```

Comment:
What do you mean by 'multiple checkpoints'?

Reply:
Just pushed a commit which should clarify this. I mean that we checkpoint on each interaction from users (and so getting multiple if we assume that user interactions are spread across the week)